metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -0,0 +1,1152 @@
+
+  override def implementation(
+      position: TextDocumentPositionParams
+  ): CompletableFuture[ju.List[Location]] =
+    getServiceFor(position.getTextDocument().getUri()).implementation(position)
+
+  override def hover(params: HoverExtParams): CompletableFuture[Hover] =
+    getServiceFor(params.textDocument.getUri()).hover(params)
+
+  override def documentHighlights(
+      params: TextDocumentPositionParams
+  ): CompletableFuture[ju.List[DocumentHighlight]] =
+    getServiceFor(params.getTextDocument.getUri()).documentHighlights(params)
+
+  override def documentSymbol(params: DocumentSymbolParams): CompletableFuture[
+    messages.Either[ju.List[DocumentSymbol], ju.List[SymbolInformation]]
+  ] =
+    getServiceFor(params.getTextDocument.getUri()).documentSymbol(params)
+
+  override def formatting(
+      params: DocumentFormattingParams
+  ): CompletableFuture[ju.List[TextEdit]] =
+    getServiceFor(params.getTextDocument.getUri()).formatting(params)
+
+  override def onTypeFormatting(
+      params: DocumentOnTypeFormattingParams
+  ): CompletableFuture[ju.List[TextEdit]] =
+    getServiceFor(params.getTextDocument.getUri()).onTypeFormatting(params)
+
+  override def rangeFormatting(
+      params: DocumentRangeFormattingParams
+  ): CompletableFuture[ju.List[TextEdit]] =
+    getServiceFor(params.getTextDocument.getUri()).rangeFormatting(params)
+
+  override def prepareRename(
+      params: TextDocumentPositionParams
+  ): CompletableFuture[lsp4j.Range] =
+    getServiceFor(params.getTextDocument.getUri()).prepareRename(params)
+
+  override def rename(params: RenameParams): CompletableFuture[WorkspaceEdit] =
+    getServiceFor(params.getTextDocument.getUri()).rename(params)
+
+  override def references(
+      params: ReferenceParams
+  ): CompletableFuture[ju.List[Location]] =
+    getServiceFor(params.getTextDocument.getUri()).references(params)
+
+  override def prepareCallHierarchy(
+      params: CallHierarchyPrepareParams
+  ): CompletableFuture[ju.List[CallHierarchyItem]] =
+    getServiceFor(params.getTextDocument.getUri()).prepareCallHierarchy(params)
+
+  override def callHierarchyIncomingCalls(
+      params: CallHierarchyIncomingCallsParams
+  ): CompletableFuture[ju.List[CallHierarchyIncomingCall]] =
+    getServiceFor(params.getItem.getUri).callHierarchyIncomingCalls(params)
+
+  override def callHierarchyOutgoingCalls(
+      params: CallHierarchyOutgoingCallsParams
+  ): CompletableFuture[ju.List[CallHierarchyOutgoingCall]] =
+    getServiceFor(params.getItem.getUri).callHierarchyOutgoingCalls(params)
+
+  override def completion(
+      params: CompletionParams
+  ): CompletableFuture[CompletionList] =
+    getServiceFor(params.getTextDocument.getUri).completion(params)
+
+  override def completionItemResolve(
+      item: CompletionItem
+  ): CompletableFuture[CompletionItem] = {
+    val res =
+      for {
+        data <- item.data
+      } yield getServiceFor(data.folderUri).completionItemResolve(item)
```

Comment:
Let's just use service for the current folder. You can't switch the file without closing completion window